### DISCUSSION TITLE
[Bug?] Why is the value of isValid wrong when the controller is included in the dynamically displayed input item when dealing with nested values?

### DISCUSSION THREAD

**Discussion by 2no** [react-hook-form/react-hook-form#9277] [Q&A]
Why does it behave this way?

https://codesandbox.io/s/kind-snowflake-cwhz7g?file=/src/App.tsx

### Step

1. Enter text in the input field under `Nested inputs`.
2. Click `Visible`.
3. isValid: `true` or `false`

### Expected behavior

The input field displayed by clicking Visible contains a required field, so isValid is false.

This problem only occurs with nested values.

https://user-images.githubusercontent.com/305405/198255514-22dddb27-d13a-4a53-ad8f-694a3700e18a.mov

**Comment by leapful** [comment]
@2no 

The issue might be related to method `executeBuiltInValidation` from file "createFormControl.ts" where `context.valid` is being set to valid state (true or false) of last nested field because of recursion.

@bluebill1049  Since we are using recursive to validate nested field, can we change to `return context.valid` at line 437 ?

https://github.com/react-hook-form/react-hook-form/blob/8dbf4dd8c958205a3a020d42c057c6750de382dc/src/logic/createFormControl.ts#L417-L462

**Reply by bluebill1049** [reply]
yes, we can @leapful feel free to send a PR 🙏
